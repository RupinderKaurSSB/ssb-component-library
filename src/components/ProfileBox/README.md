ProfileBox
========

> Wrapper for Link and Text

### Usage

#### HTML

```html
<div class="ssb-profile-box">
    <a class="ssb-link header" href=" " target="" rel="">
        <span class="link-text">This is a title link</span>
    </a>
    <p class="ssb-paragraph ">Explaining text about something.</p>
</div>
```

#### React

```jsx harmony
<ProfileBox link=" " text="Explaining text about something." title="This is a title link" />
```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| link | string | Destination for navigation |
| text | string | Renders text |
| title | string | Link text  |
