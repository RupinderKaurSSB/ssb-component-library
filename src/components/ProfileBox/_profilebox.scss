.ssb-profile-box {
  background: $ice-cold-grey;
  display: inline-block;
  flex-direction: column;
  padding: $global-padding-size * 2;
}
