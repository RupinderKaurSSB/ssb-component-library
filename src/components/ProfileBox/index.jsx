import React from 'react';
import PropTypes from 'prop-types';
import Link from '../Link';
import Paragraph from '../Paragraph';

const ProfileBox = ({
	link, text, title,
}) => (
	<div className="ssb-profile-box">
		<Link href={link} linkType="header">{title}</Link>
		<Paragraph>{text}</Paragraph>
	</div>
);

ProfileBox.defaultProps = {};

ProfileBox.propTypes = {
	link: PropTypes.string,
	text: PropTypes.string,
	title: PropTypes.string,
};

export default ProfileBox;
