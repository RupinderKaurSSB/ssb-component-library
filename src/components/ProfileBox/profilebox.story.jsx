import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import ProfileBox from './index';

storiesOf('ProfileBox', module).addDecorator(centered)
	.add('Default', () => (
		<ProfileBox link=" " text="Explaining text about something." title="This is a title link" />
	))
	.add('Three boxes', () => (
		<div style={{ display: 'grid', gridTemplateColumns: 'repeat(3, 250px)', gridColumnGap: '20px' }}>
			<ProfileBox link=" " text="Explaining text about something." title="This is a title link Box 1" />
			<ProfileBox link=" " text="Explaining text about something." title="This is a title link Box 2" />
			<ProfileBox link=" " text="Explaining text about something." title="This is a title link Box 3" />
		</div>
	));
