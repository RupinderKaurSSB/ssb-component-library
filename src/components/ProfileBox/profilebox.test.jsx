import React from 'react';
import {shallow} from 'enzyme';
import ProfileBox from './index';

describe('ProfileBox component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<ProfileBox>ProfileBox</ProfileBox>);
		expect(wrapper).toMatchSnapshot();
	});
	test('Toggles classNames correctly', () => {
		const wrapper = shallow(<ProfileBox centered>ProfileBox</ProfileBox>);
		expect(wrapper.find('.profile-content').hasClass('centered')).toEqual(true);
	});

});
