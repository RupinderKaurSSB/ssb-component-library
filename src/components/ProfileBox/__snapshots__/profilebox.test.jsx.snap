// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileBox component Matches the snapshot 1`] = `
<div
  className="ssb-profile-box"
>
  <button
    className="profile-content"
    onClick={[Function]}
  >
    ProfileBox
  </button>
</div>
`;
